compressors/integer_coders.py

```diff
@@ -85,3 +86,149 @@ def test_universal_uint_encode_decode():
 
     is_lossless, _ = try_lossless_compression(data_block, encoder, decoder)
     assert is_lossless
+
+
+"""
+shared initialization for Golomb encoder and decoder
+"""
+
+
+def _initialize_golomb_coder(coder, M: int):
+    assert M > 0
+    coder.M = M
+    coder.b = int(math.floor(math.log2(coder.M)))
+    coder.cutoff = 2 ** (coder.b + 1) - coder.M
+    if coder.cutoff == coder.M:
+        coder.rice_code = True
+    else:
+        coder.rice_code = False
+
+
+class GolombUintEncoder(DataEncoder):
+    """
+    Golomb code with parameter M based on
+    https://en.wikipedia.org/wiki/Golomb_coding#Simple_algorithm:
```

Comment:
Probably add a bit more of detail on the Golomb code algorithm here.

Reply:
Please use the google docstring style (probably add an add on the VSCode/atom) .. also please follow the same for the functions

Reply:
Thanks, will do. Also see my comment below in another response:

> If we could add the formatting expectations somewhere (Readme or in PR guidelines), that would be good. I just ran the black thing for now

Reply:
Yes ll do that

Reply:
https://gist.github.com/redlotus/3bc387c2591e3e908c9b63b97b11d24e FYI